[wasm] Support parameters of type Enum with icalls

maraf opened the pull request:
Support for reading signatures for icalls with overloaded parameters containing enums.

Fixes https://github.com/dotnet/runtime/issues/61277

maraf committed:
[wasm] Support parameters of type Enum with icalls

maraf committed:
Test

maraf commented:
/azp run runtime-wasm

radical commented on src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:
```diff
@@ -115,6 +116,120 @@ public static int Main()
+            </Project>
+            """;
+
+            string AddAssembly(string name) => $"<WasmPInvokeAssembly Include=\"{Path.Combine(libraryDir, "bin", "Debug", "net7.0", "browser-wasm", name + ".dll")}\" />";
```
`net7.0` -> `BuildTestBase.DefaultTargetFramework`

radical commented on src/tests/BuildWasmApps/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs:
```diff
@@ -115,6 +116,120 @@ public static int Main()
+                    AssertAppBundle: false
+                )
+            );
+
```
```suggestion
```

maraf committed:
Make the paths less hardcoded.

Run the test only when IsUsingWorkloads=true.
Add comments about projects being build.

maraf commented:
/azp run runtime-wasm

maraf commented:
@radical I guess that the packs path is not correct on linux

```
error MSB4062: The "ManagedToNativeGenerator" task could not be loaded from the assembly /datadisks/disk1/work/B31A0A2B/w/B96F09C5/e/snfgt5dd.gketasks/./dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/tasks/net7.0/WasmAppBuilder.dll. Could not load file or assembly '/datadisks/disk1/work/B31A0A2B/w/B96F09C5/e/snfgt5dd.gketasks/dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/tasks/net7.0/WasmAppBuilder.dll'. The system cannot find the file specified.
error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.
```

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-73817-merge-807125bd1ba34ecbb8/Workloads-.PInvokeTableGeneratorTests/1/console.b273b450.log?helixlogtype=result

maraf committed:
Fix packs path on linux.

maraf commented:
/azp run runtime-wasm

radical committed:
Merge branch 'main' into WasmIcallEnum

radical committed:
[wasm] WBT: convert sdk path to full, if needed

radical committed:
minor cleanup

radical commented:
`Libraries Test Run release coreclr Linux_musl x64 Debug` failure is https://github.com/dotnet/runtime/issues/73688 .
